storage/namespace.go

```diff
@@ -568,9 +568,7 @@ func (n *dbNamespace) Flush(
 func (n *dbNamespace) NeedsFlush(startInclusive time.Time, endInclusive time.Time) bool {
 	var (
 		blockSize   = n.nopts.RetentionOptions().BlockSize()
-		startBlock  = startInclusive.Truncate(blockSize)
-		endBlock    = endInclusive.Truncate(blockSize)
-		blockStarts = timesInRange(startBlock, endBlock, blockSize)
+		blockStarts = timesInRange(startInclusive, endInclusive, blockSize)
```

Comment:
Okay sounds good, even better we could name the paramaters something like `alignedStart` and `alignedEnd` so it's much easier to tell that the arguments should always be aligned.